cmake/CMakeLists.txt

```diff
@@ -520,9 +520,12 @@ include_directories(${LAMMPS_STYLE_HEADERS_DIR})
 ###########################################
 # Actually add executable and lib to build
 ############################################
+set(LAMMPS_SUFFIX "" CACHE STRING "Suffix to append to lmp binary and liblammps")
```

Comment:
This could be done by the unreadable expression `$<$<BOOL:${LAMMPS_MACHINE}>:_>`, e.g.:
```diff
diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 3582fe96a..650308dc9 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -525,7 +525,7 @@ mark_as_advanced(LAMMPS_MACHINE)

 add_library(lammps ${LIB_SOURCES})
 target_link_libraries(lammps ${LAMMPS_LINK_LIBS})
-set_target_properties(lammps PROPERTIES SOVERSION ${SOVERSION} OUTPUT_NAME lammps${LAMMPS_MACHINE})
+set_target_properties(lammps PROPERTIES SOVERSION ${SOVERSION} OUTPUT_NAME lammps$<$<BOOL:${LAMMPS_MACHINE}>:_>${LAMMPS_MACHINE})
 if(INSTALL_LIB)
   install(TARGETS lammps LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   install(FILES ${LAMMPS_SOURCE_DIR}/lammps.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
@@ -535,7 +535,7 @@ endif()

 add_executable(lmp ${LMP_SOURCES})
 target_link_libraries(lmp lammps)
-set_target_properties(lmp PROPERTIES OUTPUT_NAME lmp${LAMMPS_MACHINE})
+set_target_properties(lmp PROPERTIES OUTPUT_NAME lmp$<$<BOOL:${LAMMPS_MACHINE}>:_>${LAMMPS_MACHINE})
 install(TARGETS lmp DESTINATION ${CMAKE_INSTALL_BINDIR})
 if(ENABLE_TESTING)
   add_test(ShowHelp lmp${LAMMPS_MACHINE} -help)
```